### PR TITLE
Skill - BarricadePlus

### PR DESCRIPTION
**Pull request by JesterWizard**
Reduce enemy damage of every attack after the first in the battle by 50% each time. 

So if the enemy attacks 4 times a battle for 16 damage, then the totals will looks like this 16 -> 8 -> 4 -> 2

### CODE REVIEW

File: Wizardry/Core/BattleSys/Source/BattleHit.c [hunk 1]
```diff
@@ -569,8 +569,34 @@ STATIC_DECLAR int BattleHit_CalcDamage(struct BattleUnit * attacker, struct Batt
...
+            barricadePlus_activated = true;
+            int reduction_amount = SKILL_EFF0(SID_BarricadePlus);
+
+            if (act_flags->round_cnt_hit > 2)
+            {
+                int halved_value = reduction_amount;
+                
+                for (int i = 0; i < act_flags->round_cnt_hit - 2; i++)
+                {
+                    halved_value = halved_value / 2;
```
Comment (MokhaLeee) [inline Wizardry/Core/BattleSys/Source/BattleHit.c:588]: it seems unused

File: Wizardry/Core/BattleSys/Source/BattleHit.c [hunk 2]
```diff
@@ -569,8 +569,34 @@ STATIC_DECLAR int BattleHit_CalcDamage(struct BattleUnit * attacker, struct Batt
         decrease += DAMAGE_DECREASE(SKILL_EFF0(SID_CrusaderWard));
 #endif
 
+    bool barricadePlus_activated = false;
```
Comment (MokhaLeee) [inline Wizardry/Core/BattleSys/Source/BattleHit.c:572]: don't define a value inside blocks

File: Wizardry/Core/BattleSys/Source/BattleHit.c [hunk 3]
```diff
@@ -569,8 +569,34 @@ STATIC_DECLAR int BattleHit_CalcDamage(struct BattleUnit * attacker, struct Batt
...
+
+#if (defined(SID_BarricadePlus) && (COMMON_SKILL_VALID(SID_BarricadePlus)))
+    if (BattleSkillTester(defender, SID_BarricadePlus))
+    {
+        if (defender->unit.curHP < defender->hpInitial)
```
Comment (MokhaLeee) [inline Wizardry/Core/BattleSys/Source/BattleHit.c:577]: seems no need on such judgement

File: Wizardry/Core/BattleSys/Source/BattleHit.c [hunk 4]
```diff
@@ -569,8 +569,34 @@ STATIC_DECLAR int BattleHit_CalcDamage(struct BattleUnit * attacker, struct Batt
...
+        {
+            barricadePlus_activated = true;
+            int reduction_amount = SKILL_EFF0(SID_BarricadePlus);
+
+            if (act_flags->round_cnt_hit > 2)
+            {
+                int halved_value = reduction_amount;
+                
+                for (int i = 0; i < act_flags->round_cnt_hit - 2; i++)
```
Comment (MokhaLeee) [inline Wizardry/Core/BattleSys/Source/BattleHit.c:586]: the same issue. value should not define inside

File: Wizardry/Core/BattleSys/Source/BattleHit.c (hunk 3 above)
Comment (JesterWizard) [reply Wizardry/Core/BattleSys/Source/BattleHit.c:577]: True, since we're already checking for the number of attacks the skill holder has sustained.

**Comment by MokhaLeee** [comment]
fine, I may update later

**Pull request merged**